src/runtime/SrcDocIFrameController.ts

```diff
@@ -104,6 +106,31 @@ export class SrcDocIFrameController extends Disposable implements IEmbeddedConte
         return this.executeOnController((controller) => controller.setStyles(styles));
     }
 
+    private scrollToHash() {
+        const hash = window.location.hash.substring(1);
+
+        if (hash) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const document = this.host.contentWindow!.document;
+            const element = document.getElementById(hash);
+            if (element) {
+                element.scrollIntoView({behavior: 'smooth'});
+            }
+        }
+    }
+
+    private async handleHashChange() {
+        // цait until all iframes managed by the parent controller are fully loaded,
+        // and parent containers have heights properly adjusted.
+        setTimeout(() => {
+            this.scrollToHash();
+        }, PARENT_LOADED_AND_RESIZED_TIMEOUT);
+
+        const handleHashChange = () => this.scrollToHash();
+        window.addEventListener('hashchange', handleHashChange);
+        this.dispose.add(() => window.removeEventListener('hashchange', handleHashChange));
+    }
```

Comment:
@brotheroftux @d3m1d0v 

The current version works fine, but has several issues:
1.	addEventListener('hashchange') is added in every iframe on the page, whereas it could be set just once in the EmbeddedIFrameController.
1.	A timeout had to be added because I cannot ensure, within the iframe, that all other iframes in its parent are fully loaded and that the parent’s height has been fully adjusted.
1.	If multiple iframes contain the same anchor, the scroll might happen to an iframe that is not currently visible on the screen. Ideally, scrolling should happen to the first found anchor across all iframes at the EmbeddedIFrameController level.
1.	I also realized that the addAnchorLinkHandlers method within the iframe won’t work correctly if the link points to a different iframe on the page.

Therefore, I suggest discussing a possible future refactor:
1.	The root controller should provide a loading status—for example, right now we can get information about the loading status of all iframes here, but the result of Promise.all is not utilized anywhere:
[EmbeddedContentRootController.ts#L84](https://github.com/diplodoc-platform/html-extension/blob/2ebcb4f5ef9e5683c433d58b9dde4ccdcb278f1e/src/runtime/EmbeddedContentRootController.ts#L84)
1.	Ideally, we should also ensure that the parent container has finalized the height of all its iframes. (I’ve tried various approaches with ResizeObserver, but haven’t found a clean solution yet.)
1.	I would prefer not to overload the EmbeddedContentRootController with link-handling logic. Instead, this should be configurable via an additional argument or method. There’s already a forEach method, so this could be something like executeOnRootController.
1. It will also be necessary to somehow dispatch clicks on # links from child iframes to handle them in the root controller (because the link may refer to the header in another iframe). I’m leaning towards something similar to [this approach](https://github.com/diplodoc-platform/html-extension/blob/2ebcb4f5ef9e5683c433d58b9dde4ccdcb278f1e/src/iframe/IFrameController.ts#L31).

What do you think of each of the things in the proposals?